File: app/src/main/java/com/azizbek/todolist/screens/main/MainViewModel.java
package com.azizbek.todolist.screens.main;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.azizbek.todolist.App;
import com.azizbek.todolist.model.Note;

import java.util.List;

public class MainViewModel extends ViewModel {
    private LiveData<List<Note>> noteLiveData = App.getInstance().getNoteDao().getAllLiveData();

    public LiveData<List<Note>> getNoteLiveData() {
        return noteLiveData;
    }

    private LiveData<List<Note>> progressTasks = App.getInstance().getNoteDao().getAllProgressTasks();

    public LiveData<List<Note>> getProgressTasks() {
        return progressTasks;
    }

    private LiveData<List<Note>> completedTasks = App.getInstance().getNoteDao().getAllCompletedTasks();

    public LiveData<List<Note>> getCompletedTasks() {
        return completedTasks;
    }

}

File: app/src/main/java/com/azizbek/todolist/screens/main/MainViewModel.kt
package com.azizbek.todolist.screens.main

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.azizbek.todolist.App
import com.azizbek.todolist.data.NoteDao
import com.azizbek.todolist.model.Note
import com.azizbek.todolist.repository.ToDoRepository

class MainViewModel : ViewModel() {
    private val toDoRepository:ToDoRepository
    val noteLiveData: LiveData<List<Note>>
    val progressTasks: LiveData<List<Note>>
    val completedTasks: LiveData<List<Note>>

    init {
        val noteDao=App.getInstance().noteDao
        toDoRepository=ToDoRepository(noteDao)
        noteLiveData=toDoRepository.allLiveData
        progressTasks=toDoRepository.allProgressTasks
        completedTasks=toDoRepository.allCompletedTasks
    }


}